gtsam/sfm/DsfTrackGenerator.h

```diff
@@ -0,0 +1,217 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file DsfTrackGenerator.h
+ * @date July 2022
+ * @author John Lambert
+ * @brief Identifies connected components in the keypoint matches graph.
+ */
+
+#pragma once
+#include <Eigen/Core>
+
+#include <gtsam/base/DSFMap.h>
+#include <gtsam/geometry/Point2.h>
+
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <optional>
+#include <vector>
+
+namespace gtsam {
+
+typedef DSFMap<IndexPair> DSFMapIndexPair;
+typedef Eigen::MatrixX2i CorrespondenceIndices; // N x 2 array
+
+//struct Keypoints;
+using KeypointCoordinates = Eigen::MatrixX2d;
+
+
+// Output of detections in an image.
+// Coordinate system convention:
+// 1. The x coordinate denotes the horizontal direction (+ve direction towards the right).
+// 2. The y coordinate denotes the vertical direction (+ve direction downwards).
+// 3. Origin is at the top left corner of the image.
+struct Keypoints {
+
+  // The (x, y) coordinates of the features, of shape Nx2.
+  KeypointCoordinates coordinates;
+
+  // Optional scale of the detections, of shape N.
+  // Note: gtsam::Vector is typedef'd for Eigen::VectorXd.
+  boost::optional<gtsam::Vector> scales;
+
+  /// Optional confidences/responses for each detection, of shape N.
+  boost::optional<gtsam::Vector> responses;
+
+  Keypoints(const gtsam::KeypointCoordinates& coordinates): coordinates(coordinates) {}; // boost::none
+};
+
+using KeypointsList = std::vector<Keypoints>;
```

Comment:
KeypointsVector

Reply:
Thanks, updated KeypointsList to KeypointsVector.